tests/test_scalars.py

```diff
@@ -15,26 +22,568 @@
+
+    as_pd_timedelta = pd.Timedelta(days=1)
+    as_dt_timedelta = dt.timedelta(days=1)
+    as_offset = 3 * Day()
+    as_timedelta_index = pd.TimedeltaIndex([1, 2, 3], "D")
+    as_timedelta_series = pd.Series(as_timedelta_index)
+    check(assert_type(as_timedelta_series, TimedeltaSeries), pd.Series)
+    as_np_ndarray_td64 = np_td64_arr
+
+    check(assert_type(ts + as_pd_timedelta, pd.Timestamp), pd.Timestamp)
+    check(assert_type(as_pd_timedelta + ts, pd.Timestamp), pd.Timestamp)
+
+    check(assert_type(ts + as_dt_timedelta, pd.Timestamp), pd.Timestamp)
+    check(assert_type(as_dt_timedelta + ts, pd.Timestamp), pd.Timestamp)
+
+    check(assert_type(ts + as_offset, pd.Timestamp), pd.Timestamp)
+    check(assert_type(as_offset + ts, pd.Timestamp), pd.Timestamp)
+
+    check(assert_type(ts + as_timedelta_index, pd.DatetimeIndex), pd.DatetimeIndex)
+    check(assert_type(as_timedelta_index + ts, pd.DatetimeIndex), pd.DatetimeIndex)
+
+    check(assert_type(ts + as_timedelta_series, "TimestampSeries"), pd.Series)
+    check(assert_type(as_timedelta_series + ts, "TimestampSeries"), pd.Series)
+
+    check(assert_type(ts + as_np_ndarray_td64, npt.NDArray[np.datetime64]), np.ndarray)
+    # pyright and mypy disagree on the type of this expression
+    # pyright: Any, mypy: npt.NDArray[np.datetime64]
+    check(
+        assert_type(
+            as_np_ndarray_td64 + ts,  # pyright: ignore [reportGeneralTypeIssues]
+            npt.NDArray[np.datetime64],
+        ),
+        np.ndarray,
+    )
+
+    # Reverse order is not possible for all of these
+    check(assert_type(ts - as_pd_timedelta, pd.Timestamp), pd.Timestamp)
+    check(assert_type(ts - as_dt_timedelta, pd.Timestamp), pd.Timestamp)
+    check(assert_type(ts - as_offset, pd.Timestamp), pd.Timestamp)
+    check(assert_type(ts - as_timedelta_index, pd.DatetimeIndex), pd.DatetimeIndex)
+    check(assert_type(ts - as_timedelta_series, "TimestampSeries"), pd.Series)
+    check(assert_type(ts - as_np_ndarray_td64, npt.NDArray[np.datetime64]), np.ndarray)
+
+
+def test_timestamp_cmp() -> None:
+    ts = pd.Timestamp(year=2000, month=3, day=24, hour=12, minute=27)
+
+    np_dt64_arr: npt.NDArray[np.datetime64] = np.array(
+        [1, 2, 3], dtype="datetime64[ns]"
+    )
+
+    c_timestamp = ts
+    c_np_dt64 = np.datetime64(1, "ns")
+    c_dt_datetime = dt.datetime(year=2000, month=1, day=1)
+    c_datetimeindex = pd.DatetimeIndex(["2000-1-1"])
+    c_np_ndarray_dt64 = np_dt64_arr
+    # Typing provided since there is no way to get a Series[Timestamp],
+    # which is a different type from a TimestampSeries
+    c_series_dt64: pd.Series[pd.Timestamp] = pd.Series(
+        [1, 2, 3], dtype="datetime64[ns]"
+    )
+    c_series_timestamp = pd.Series(pd.DatetimeIndex(["2000-1-1"]))
+    check(assert_type(c_series_timestamp, TimestampSeries), pd.Series)
+    # Use xor to ensure one is True and the other is False
```

Comment:
don't understand this comment.  Where are you using `xor` ??

Reply:
In python when you know `a` and `b` are bool, then  `a != b` is the xor operator.  It ensures that 1 and only 1 is True and avoids directly testing pandas.

Reply:
OK, I never thought of `!=` as `xor` !